beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/historical/HistoricalBlockSyncService.java

```diff
@@ -123,9 +123,11 @@ public static HistoricalBlockSyncService create(
       final CombinedChainDataClient chainData,
       final AsyncBLSSignatureVerifier signatureVerifier,
       final SyncStateProvider syncStateProvider,
-      final boolean reconstructHistoricStatesEnabled) {
+      final boolean reconstructHistoricStatesEnabled,
+      final Optional<String> genesisStateResource) {
     ReconstructHistoricalStatesService reconstructHistoricalStatesService =
-        new ReconstructHistoricalStatesService();
+        new ReconstructHistoricalStatesService(
+            storageUpdateChannel, chainData, spec, genesisStateResource);
```

Comment:
I wonder if we should only create this service if reconstructing states is enabled. Then instead of passing the service and a boolean through, we'd just pass `Optional<ReconstructHistoricalStatesService>` - if it's present it's enabled, if it's absent it isn't.